resteasy-core/src/main/java/org/jboss/resteasy/statistics/MethodStatisticsLoggerImpl.java

```diff
@@ -0,0 +1,75 @@
+package org.jboss.resteasy.statistics;
+
+import org.jboss.resteasy.spi.statistics.MethodStatisticsLogger;
+
+public class MethodStatisticsLoggerImpl implements MethodStatisticsLogger {
+   private long invocationCnt = 0;
+   private long failureCnt = 0;
+   private long totalExecutionTime = 0;
+
+   @Override
+   public long timestamp(){
+      return System.nanoTime();
+   }
+
+   @Override
+   public void duration(final long fromTimestamp) {
+      // invocation count and execution time are related.
+      // Set them together so avgExecutionTime will be calculated correctly
+      synchronized (this)
+      {
+         ++invocationCnt;
+         totalExecutionTime += (System.nanoTime() - fromTimestamp);
+      }
+   }
+
+   @Override
+   public void incFailureCnt() {
+      ++failureCnt;
+   }
+
+   @Override
+   public void reset() {
+      synchronized (this)
+      {
+         invocationCnt = 0;
+         failureCnt = 0;
+         totalExecutionTime = 0;
+      }
+   }
+
+   @Override
+   public long getInvocationCnt() {
+      return invocationCnt;
+   }
+
+   @Override
+   public long getFailedInvocationCnt() {
+      return failureCnt;
+   }
+
+   @Override
+   public long getAvgExecutionTime() {
+      long avgExecTime = -1;
+
+      try
+      {
+         synchronized (this)
+         {
+            avgExecTime = totalExecutionTime / invocationCnt;
+         }
```

Comment:
Why is the synchronized block here within the 'try' part of the try-catch block? Shouldn't the 'catch' part also run in the synchronized block ?